test/test_tagging.py

```diff
@@ -0,0 +1,110 @@
+import json
+from unittest.mock import (
+    PropertyMock,
+    patch,
+)
+
+from azul.deployment import (
+    populate_tags,
+)
+from azul.logging import (
+    configure_test_logging,
+)
+from azul_test_case import (
+    AzulUnitTestCase,
+)
+
+
+def setupModule():
+    configure_test_logging()
+
+
+class TestTagging(AzulUnitTestCase):
+
+    @patch('subprocess.run', new_callable=PropertyMock)
+    def test(self, terraform_mock):
+        # We mock Terraform since it's not available on Travis. The mocked
+        # response only includes resources used by this test case, so we don't
+        # need to update it manually.
+        terraform_mock.return_value.stdout = json.dumps({
+            'format_version': '0.1',
+            'provider_schemas': {
+                'aws': {
+                    'resource_schemas': {
+                        'aws_vpc': {
+                            'block': {
+                                'attributes': {
+                                    'tags': {}
+                                }
+                            }
+                        }
+                    }
+                },
+                'gcp': {
+                    'resource_schemas': {
+                        'google_compute_instance': {
+                            'block': {
+                                'attributes': {
+                                    'tags': {}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }).encode()
+
+        tagged_aws_resource = {
+            'resource': {
+                'aws_vpc': {
+                    'name': {}
+                }
+            }
+        }
+        res = populate_tags(tagged_aws_resource)
+        tags = res['resource'][0]['aws_vpc'][0]['name']['tags']
+        self.assertIn('Name', tags)
```

Comment:
Wouldn't it be more strict and easier to comprehend (albeit more verbose) to assert equality to a complete literal?

Reply:
Tag values can't be hardcoded because they vary depending on stage name, user, etc. I think the code can be clearer and more strict, though. Should I:
* Replicate the logic in azul.deployment._tags in the test code
* Provide a complete "expected" literal and compare only the keys of each dictionary (which would be sufficient and also more strict)
* Something else?

Reply:
I've pushed a commit that uses option #2. I'll wait until you see this before requesting review.